core/src/main/java/org/infinispan/commands/write/SinglePutKeyValueCommand.java

```diff
@@ -0,0 +1,307 @@
+package org.infinispan.commands.write;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+import org.infinispan.atomic.CopyableDeltaAware;
+import org.infinispan.atomic.Delta;
+import org.infinispan.atomic.DeltaAware;
+import org.infinispan.commands.CommandInvocationId;
+import org.infinispan.commands.FlagAffectedCommand;
+import org.infinispan.commands.Visitor;
+import org.infinispan.commands.remote.CacheRpcCommand;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.util.EnumUtil;
+import org.infinispan.container.entries.MVCCEntry;
+import org.infinispan.context.InvocationContext;
+import org.infinispan.lifecycle.ComponentStatus;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.util.ByteString;
+import org.infinispan.util.concurrent.locks.RemoteLockCommand;
+
+/**
+ * Command for a replicable, non-tx, cache.put(k, v) call that uses
+ * default metadata and has no listeners attached.
+ *
+ * - A replicable, non-tx put does not need to be wrapped in another command, it can be sent directly.
+ * - Single metadata is default, no metadata information needs to be serialized.
+ * - A non-conditional put always matches value, so is always successful.
+ * - If value equivalence is default, no need to serialize that info.
+ * - Only ignore return values flag set, so no need to serialize that info.
+ */
+public final class SinglePutKeyValueCommand
+   implements
+      DataWriteCommand, RemoteLockCommand, FlagAffectedCommand,
+      AdvancedExternalizer<SinglePutKeyValueCommand>,
```

Comment:
Ok, now I understand why you made it directly the externalizer. Though, if you want to go this direction, I would prefer implementing just `AdvancedExternalizable` interface that would return an instance (singleton) of the externalizer.